documentation/writing_yaml_plans.md

```diff
@@ -77,40 +77,37 @@ Each plan name segment must begin with a lowercase letter and:
 
 ## Plan structure
 
-YAML plans contain a list of steps with optional parameters and results.
+YAML plans contain a list of steps and can include additional optional keys.
+The following top-level keys are available:
 
-YAML maps accept these keys:
+| Key | Type | Description | Required |
+| --- | --- | --- | --- |
+| `description` | `String` | The plan description. Appears in `bolt plan show <PLAN NAME>` and `Get-BoltPlan <PLAN NAME>` output. | |
+| `parameters` | `Hash` | A hash of plan parameters. Each key is the name of the parameter and the value is the parameter definition. | |
+| `private` | `Boolean` | Whether the plan should appear in `bolt plan show` and `Get-BoltPlan` output. | |
+| `return` | `Array`, `Boolean`, `Hash`, `Number`, `String` | The value to return from the plan. Must evaluate to a valid [PlanResult](bolt_types_reference.md#planresult). | |
+| `steps` | `Array` | The list of steps to run. | ✓ |
 
--   `steps`: The list of steps to perform
--   `parameters`: (Optional) The parameters accepted by the plan
--   `return`: (Optional) The value to return from the plan
-
-### Steps key
+## Steps
 
 The `steps` key is an array of step objects, each of which corresponds to a
 specific action to take.
 
-When the plan runs, each step is executed in order. If a step fails, the plan
-halts execution and raises an error containing the result of the step that
-failed.
-
-Steps use these fields:
-
--   `name`: (Optional) A unique name that can be used to refer to the result of
-    the step later.
--   `description`: (Optional) An explanation of what the step is doing.
+When the plan runs, each step is executed in order. By default, if a step fails
+the plan halts execution and raises an error containing the result of the step
+that failed.
```

Comment:
```suggestion
When the plan runs, each step is executed in order. If a step fails,
the plan halts execution and raises an error containing the result of the step
that failed.
```
Do we need the "by default" here? It's a bit awkward with the conditional cos you need another comma.